model/storage/hugging_face/hugging_face_model_store.py

```diff
@@ -24,14 +25,22 @@ async def upload_model(self, model: Model) -> ModelId:
             safe_serialization=True,
         )
 
-        # Return a new ModelId with the uploaded commit.
-        return ModelId(
+        model_id_with_commit = ModelId(
             namespace=model.id.namespace,
             name=model.id.name,
             hash=model.id.hash,
             commit=commit_info.oid,
         )
 
+        # TODO consider skipping the redownload if a hash is already provided.
+        # To get the hash we need to redownload it at a local tmp directory after which it can be deleted.
+        tmp_path = "sn9_tmp"
```

Comment:
Should this live in /tmp/?

Reply:
Done. Using tempfile module as suggested.